### ISSUE TITLE
0.9.9: test_all fails

### ISSUE THREAD

**Issue by olebole** [astropy/montage-wrapper#51]
Hi,

when trying to build 0.9.9 (which is, BTW, on pypi but not on github) for Debian, I get a failure in `test_all`:
```
___________________________________ test_all ___________________________________

    def test_all():
    
>       stat = mArchiveList('2MASS', 'k', 'M31', 0.2, 0.2, 'm31_2mass_k.tbl')

montage_wrapper/tests/test_commands.py:10: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
montage_wrapper/commands.py:284: in mArchiveList
    return status.parse_struct("mArchiveList", p.stdout.read().strip())
montage_wrapper/status.py:33: in parse_struct
    result = Struct(command, string)
montage_wrapper/status.py:67: in __init__
    self.__dict__[key] = simplify(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = '"Invalid URL string (must start \'http://\''

    def simplify(value):
        try:
            return int(value)
        except:
>           return float(value)
E           ValueError: could not convert string to float: "Invalid URL string (must start 'http://'

montage_wrapper/status.py:12: ValueError
```

I also get a number of warnings:
```
  [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
```
```
lib.linux-x86_64-2.7/montage_wrapper/tests/test_wrappers.py::TestMosaic::()::test_mosaic
  /usr/lib/python2.7/dist-packages/astropy/utils/decorators.py:482: AstropyDeprecationWarning: "clobber" was deprecated in version 2.0 and will be removed in a future version. Use argument "overwrite" instead.
    AstropyDeprecationWarning)
```
and
```
  /usr/lib/python2.7/dist-packages/astropy/io/ascii/ui.py:706: AstropyDeprecationWarning: /tmp/montage_mosaic_raw_Mtv1KJ/images_raw_all.tbl already exists. Automatically overwriting ASCII files is deprecated. Use the argument 'overwrite=True' in the future.
    output), AstropyDeprecationWarning)
```

**Comment by astrofrog** [comment]
Which version of Montage are you using?

**Comment by olebole** [comment]
The latest: Montage 5.0. montage-wrapper 0.9.8 worked fine with montage-5.0 after a few adjustments.

**Comment by astrofrog** [comment]
@olebole - can you try and see if the following command works outside of montage-wrapper?

```
mArchiveList 2MASS k M31 0.2 0.2 table.tbl
```

**Comment by olebole** [comment]
To me, it looks quite normal:
```
$ mArchiveList 2MASS k M31 0.2 0.2 table.tbl
[struct stat="OK", count="6"]
$ wc table.tbl 
   9  185 2932 table.tbl
$ cat table.tbl 
\datatype=fitshdr
|   cntr|  ctype1|  ctype2|naxis1|naxis2|    crval1|    crval2| crpix1| crpix2|     cdelt1|     cdelt2|    crota2|     scale|          ra1|         dec1|          ra2|         dec2|          ra3|         dec3|          ra4|         dec4|                                                                                         URL|                             file|
|    int|    char|    char|   int|   int|    double|    double| double| double|     double|     double|    double|    double|       double|       double|       double|       double|       double|       double|       double|       double|                                                                                        char|                             char|
 2058338 RA---SIN DEC--SIN    512   1024  10.717694  41.333905  256.50  512.50 -0.00027778  0.00027778    0.01273    0.95490    10.8122346    41.1916596    10.6232374    41.1916280    10.6227398    41.4760728    10.8125640    41.4761044 http://irsa.ipac.caltech.edu/ibe/data/twomass/full/full/971024n/s008/image/ki0080033.fits.gz 2mass-atlas-971024n-k0080033.fits
[...]
```

**Comment by astrofrog** [comment]
@olebole - do you see the error on both Python 2 and 3?

**Comment by olebole** [comment]
Yes, this happens with Python 2 (2.7.13) and Python 3 (3.6.2).

**Comment by astrofrog** [comment]
@olebole - thanks, I'll try and reproduce it.

**Comment by olebole** [comment]
I just also tried on a plain Debian Stretch chroot (Montage 4.0, Python 2.7.13 + 3.5.3, astropy 1.3, cython 0.25.2, numpy 1.12.1) with the same result.

**Comment by astrofrog** [comment]
@olebole - I'm puzzled, I just tried with a minimal docker container and couldn't reproduce the issue:

```dockerfile
FROM debian:testing

RUN apt-get update
RUN apt-get install -y montage wget python python-numpy python-astropy gcc

RUN wget https://pypi.python.org/packages/99/99/b11b373baa08c858eab5484579dc3c27e56a2b7a7cf946952ec8dae67e81/montage-wrapper-0.9.9.tar.gz
RUN tar xvzf montage-wrapper-0.9.9.tar.gz

WORKDIR montage-wrapper-0.9.9
```

Instead of failing with the http issue you are seeing, I get a failure here:

```
        if self.stat == "ERROR":
>           raise MontageError("%s: %s" % (command, self.msg))
E           MontageError: mBestImage: No image covers this point
```

which is a known issue with Montage 4.0 and 5.0.

In any case, I think ``test_all`` has to be skipped on Montage 4.0 and 5.0 until I hear back from the Montage developers.

**Comment by olebole** [comment]
That is mystic. I can confirm that I use the same source (same MD5 sum).
When I just disable [the lines that cause my error](https://github.com/astropy/montage-wrapper/blob/3cb7d55cea45be3415e5ddeb8c0df975e76d4d90/montage_wrapper/tests/test_commands.py#L10-L16), I get another one, which looks quite similar:
```
    def test_all():
    
>       stat = mArchiveExec('m31_2mass_k.tbl')

montage_wrapper/tests/test_commands.py:10: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
montage_wrapper/commands.py:203: in mArchiveExec
    return status.parse_struct("mArchiveExec", p.stdout.read().strip())
montage_wrapper/status.py:33: in parse_struct
    result = Struct(command, string)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = msg : Table m31_2mass_k.tbl needs column 'URL' or 'url' and can optionally have columns 'fname'/'file' and pixel ranges 'imin'..'jmax'
stat : ERROR
command = 'mArchiveExec'
string = 'stat=bb1ca97ec761fc37101737ba0aa2e7c5, msg=f90ebf590a5d7c2af646b71bcdd25357'

    def __init__(self, command, string):
        [...]
    
        if self.stat == "ERROR":
>           raise MontageError("%s: %s" % (command, self.msg))
E           MontageError: mArchiveExec: Table m31_2mass_k.tbl needs column 'URL' or 'url' and can optionally have columns 'fname'/'file' and pixel ranges 'imin'..'jmax'
```
Disabling the following two lines as well leads to the `MontageError` you get, but with an empty `self.msg`.
I will have a closer look (also to make sure that I didn't break something myself) in the next days.

**Comment by olebole** [comment]
OK, I found the cause for the behaviour: `mArchiveList` connects to the internet, which is not accessible during package build. To ensure that, I set the environment variable `http_proxy=127.0.0.1:9` during the build. `mArchiveList` tries to interprets this and gives the error
```
[struct stat="ERROR", msg="Invalid URL string (must start 'http://')
```
I didn't see this when I started the program from command line, since then the environment var is not set.
--> was a local problem, I am closing this. Sorry for the noise.
